src/applications/pokemon_summary_screen/window.c

```diff
@@ -232,102 +819,98 @@ void PokemonSummaryScreen_RemoveWindows(PokemonSummaryScreen *summaryScreen)
     }
 }
 
-void sub_0208FEA4(PokemonSummaryScreen *param0)
+void PokemonSummaryScreen_PrintNicknameAndGender(PokemonSummaryScreen *summaryScreen)
 {
-    Window *v0;
-    u32 v1;
-
-    v0 = &param0->staticWindows[34];
-
-    Window_FillTilemap(v0, 0);
-    Text_AddPrinterWithParamsAndColor(v0, FONT_SYSTEM, param0->monData.nickname, 0, 0, TEXT_SPEED_NO_TRANSFER, TEXT_COLOR(15, 14, 0), NULL);
+    Window *window = &summaryScreen->staticWindows[PSS_WINDOW_MON_NICKNAME_GENDER];
 
-    if (param0->monData.showGender == 0) {
-        if (param0->monData.gender == 0) {
-            MessageLoader_GetStrbuf(param0->msgLoader, 1, param0->strbuf);
-            PrintTextToWindow(param0, v0, TEXT_COLOR(3, 4, 0), PSS_ALIGNMENT_RIGHT);
-        } else if (param0->monData.gender == 1) {
-            MessageLoader_GetStrbuf(param0->msgLoader, 2, param0->strbuf);
-            PrintTextToWindow(param0, v0, TEXT_COLOR(5, 6, 0), PSS_ALIGNMENT_RIGHT);
+    Window_FillTilemap(window, 0);
```

Comment:
**thought**: Maybe we want a shared inline in `bg_window.h` like this:

```c
inline void Window_ClearTilemap(Window *window)
{
    Window_FillTilemap(window, 0);
}
```

Reply:
I really like this idea but there's already a function with that name (though admittedly one that's used very infrequently, 6 invocations in the whole repo vs 341 instances of the fill zero pattern):
```c
void Window_ClearTilemap(Window *window)
{
    sClearWindowTilemapFuncs[window->bgConfig->bgs[window->bgLayer].type](window);
}
```
Maybe something like `Window_ClearFunction` or `Window_RunClearFunc` for the function above, and then `Window_ClearTilemap` for your proposed inline?

(just more thinking, doesn't necessarily need to be answered through this PR ofc)

Reply:
Yeah, not a stopper for this PR. I think that approach would be fine.